src/qibolab/result.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from functools import cached_property
+
+import numpy as np
+import numpy.typing as npt
+
+ExecRes = np.dtype([("i", np.float64), ("q", np.float64), ("shots", np.uint32)])
+
+
+@dataclass
+class ExecutionResults:
+    """Data structure to deal with the output of execute_pulse_sequence"""
+
+    array: npt.NDArray[ExecRes]
+
+    @classmethod
+    def from_components(cls, is_, qs, shots):
+        ar = np.empty(len(is_), dtype=ExecRes)
+        ar["i"] = is_
+        ar["q"] = qs
+        ar["shots"] = shots
+        ar = np.rec.array(ar)
+        return cls(ar)
+
+    @property
+    def i(self):
+        return self.array.i
+
+    @property
+    def q(self):
+        return self.array.q
+
+    @property
+    def shots(self):
+        return self.array.shots
+
+    @cached_property
+    def msr(self):
+        """Computes msr value."""
+        return np.sqrt(self.array.i**2 + self.array.q**2)
+
+    @cached_property
+    def phase(self):
+        """Computes phase value."""
+        return np.angle(self.array.i + 1.0j * self.array.q)
+
+    @cached_property
+    def ground_state_probability(self):
+        """Computes ground state probability"""
+        return 1 - self.array.shots.mean()
+
+    def to_dict_probability(self, state=0):
```

Comment:
For compatibility with earlier versions and qibocal it may be better to have the excited state as default
```suggestion
    def to_dict_probability(self, state=1):
```
For example the probabilities returned by qblox when classifying on hardware are for the excited state.

Reply:
I see. Now it should be fixed.